x-pack/plugins/alerting/server/usage/lib/get_telemetry_from_event_log.ts

```diff
@@ -0,0 +1,620 @@
+}
+interface GetExecutionCountsAggregationBucket extends AggregationsStringTermsBucketKeys {
+  avg_execution_time: AggregationsSingleMetricAggregateBase;
+  avg_es_search_duration: AggregationsSingleMetricAggregateBase;
+  avg_total_search_duration: AggregationsSingleMetricAggregateBase;
+  execution_failures: GetExecutionCountsExecutionFailures;
+  percentile_scheduled_actions: AggregationsPercentilesAggregateBase;
+  percentile_alerts: AggregationsPercentilesAggregateBase;
+}
+
+interface IGetExecutionFailures extends AggregationsSingleBucketAggregateBase {
+  by_reason: AggregationsTermsAggregateBase<AggregationsStringTermsBucketKeys>;
+}
+
+export async function getExecutionsPerDayCount({
+  esClient,
+  eventLogIndex,
+  logger,
+}: Opts): Promise<GetExecutionsPerDayCountResults> {
+  try {
+    const generatedActionsPercentilesAgg = {
+      percentiles: {
+        field: 'kibana.alert.rule.execution.metrics.number_of_generated_actions',
+        percents: [50, 90, 99],
+      },
+    };
+
+    const alertsPercentilesAgg = {
+      percentiles: {
+        field: 'kibana.alert.rule.execution.metrics.alert_counts.active',
+        percents: [50, 90, 99],
+      },
+    };
+
+    const executionTimeAggs = {
+      avg_execution_time: {
+        avg: {
+          field: 'event.duration',
+        },
+      },
+      avg_es_search_duration: {
+        avg: {
+          field: 'kibana.alert.rule.execution.metrics.es_search_duration_ms',
+        },
+      },
+      avg_total_search_duration: {
+        avg: {
+          field: 'kibana.alert.rule.execution.metrics.total_search_duration_ms',
+        },
+      },
+    };
+
+    const query = {
+      index: eventLogIndex,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            filter: {
+              bool: {
+                must: [
+                  {
+                    term: { 'event.action': 'execute' },
+                  },
+                  {
+                    term: { 'event.provider': 'alerting' },
+                  },
+                  {
+                    range: {
+                      '@timestamp': {
+                        gte: 'now-1d',
+                      },
+                    },
+                  },
+                ],
+              },
+            },
+          },
+        },
+        aggs: {
+          ...executionTimeAggs,
+          percentile_scheduled_actions: generatedActionsPercentilesAgg,
+          percentile_alerts: alertsPercentilesAgg,
+          execution_failures: {
+            filter: {
+              term: {
+                'event.outcome': 'failure',
+              },
+            },
+            aggs: {
+              by_reason: {
+                terms: {
+                  field: 'event.reason',
+                  size: NUM_ALERTING_EXECUTION_FAILURE_REASON_TYPES,
+                },
+              },
+            },
+          },
+          by_rule_type_id: {
```

Comment:
would it make any sense to put the `by_rule_type_id` agg at the "top", and then rest underneath, so we only do the "by type" and "total" aggs once?  Presumably, we'd add up the totals from the "by type" results, since the cardinality of "by type" is fairly small.  I'm thinking it may be more performant for ES to do it that way - less total aggs to deal with.

Otherwise, the "total" and "by_rule_type_id" aggs look very similar, can we just build the JSON once and reuse it in both places?  Try reduce problems in the future where one gets changed but not the other, kinda thing.